### PR TITLE
rejecting existing applications when blacklisting

### PR DESCRIPTION
**Pull request by opqdonut**
for #2015

# Definition of Done / Review checklist

## Reviewability
- [x] link to issue
- [x] consider adding screenshots for ease of review

## Documentation
- [x] update changelog if necessary
- [x] documentation _at least_ for config options (i.e. docs folder)
- [ ] ADR for major architectural decisions or experiments

## Testing
- [x] complex logic is unit tested
- [x] valuable features are integration / browser / acceptance tested automatically

## Follow-up
- [ ] new tasks are created for pending or remaining tasks
- [ ] no critical TODOs left to implement

### CODE REVIEW

File: src/clj/rems/api/blacklist.clj [hunk 1]
```diff
@@ -56,6 +58,8 @@
       :body [command BlacklistCommand]
       :return schema/SuccessResponse
       (blacklist/add-user-to-blacklist! (getx-user-id) command)
+      (doseq [cmd (rejecter-bot/reject-all-applications-by (get-in command [:blacklist/user :userid]))]
+        (command/command! cmd))
```
Comment (Macroz) [inline src/clj/rems/api/blacklist.clj:62]: Should there be one function in the service layer that does all of this?

**Review by Macroz — APPROVED** [review]
Looks pretty good.

File: src/clj/rems/api/blacklist.clj (hunk 1 above)
Comment (opqdonut) [reply src/clj/rems/api/blacklist.clj:62]: The problem is that services.command already depends on services.blacklist, so we can't have services.blacklist depend on services.command...

But now that I look at this piece of code I notice that I should check the return value of `command!`

Comment (Macroz) [reply src/clj/rems/api/blacklist.clj:62]: Yes it's not atomic. Also it feels a bit strange to have blacklist stuff in command.

Comment (opqdonut) [reply src/clj/rems/api/blacklist.clj]: added the error handling

File: src/clj/rems/api/services/command.clj [hunk 2]
```diff
@@ -25,16 +25,17 @@
     (entitlements/update-entitlements-for-event event))
   [])
 
+;; TODO should this be in the revoke command handler instead?
```
Comment (opqdonut) [inline src/clj/rems/api/services/command.clj:28]: see my comment here

**Pull request merged**